File: app/src/main/java/com/ss/universitiesdirectory/Fragment/DetailsFragment.java
package com.ss.universitiesdirectory.Fragment;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import com.squareup.picasso.Picasso;
import com.ss.universitiesdirectory.R;

public class DetailsFragment extends Fragment {

    private View view;
    private Bundle bundle;
    private ImageView mLogo;
    private TextView mAbout;
    private Fragment fragment;
    private LinearLayout mCommunication;
    private ImageView Twitter, Facebook;
    private Button mCollage, mWebsite, mRSS, mLocation;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view =  inflater.inflate(R.layout.fragment_details, container, false);
        
        init();
        ShowData();

        return view;
    }

    private void init() {
        bundle = this.getArguments();
        mRSS = view.findViewById(R.id.RSS);
        mLogo = view.findViewById(R.id.Logo);
        Twitter = view.findViewById(R.id.Twitter);
        mAbout = view.findViewById(R.id.AboutText);
        mCollage = view.findViewById(R.id.College);
        mWebsite = view.findViewById(R.id.Website);
        Facebook = view.findViewById(R.id.Facebook);
        mLocation = view.findViewById(R.id.Location);
        mCommunication = view.findViewById(R.id.Communication);
    }

    private void ShowData() {
        if(bundle.getString("Logo") != null){
            Picasso.get().load(bundle.getString("Logo")).into(mLogo);
        }

        if(bundle.getString("About") != null){
            mAbout.setText(bundle.getString("About"));
        }

        if(bundle.getString("Website") != null){
            mWebsite.setVisibility(View.VISIBLE);
            mWebsite.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    fragment = new WebsiteFragment();
                    bundle.putString("WebsiteShow", bundle.getString("Website"));
                    fragment.setArguments(bundle);
                    FragmentTransaction transaction = getFragmentManager().beginTransaction();
                    transaction.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out, android.R.anim.fade_in, android.R.anim.fade_out);
                    transaction.replace(R.id.Container, fragment).addToBackStack(null).commit();
                }
            });
        }

        if(bundle.getString("College") != null){
            mCollage.setVisibility(View.VISIBLE);
            mCollage.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    fragment = new WebsiteFragment();
                    bundle.putString("CollegeShow", bundle.getString("College"));
                    fragment.setArguments(bundle);
                    FragmentTransaction transaction = getFragmentManager().beginTransaction();
                    transaction.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out, android.R.anim.fade_in, android.R.anim.fade_out);
                    transaction.replace(R.id.Container, fragment).addToBackStack(null).commit();
                }
            });
        }

        if(bundle.getString("News") != null){
            mRSS.setVisibility(View.VISIBLE);
            mRSS.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    fragment = new NewsFragment();
                    bundle.putString("News", bundle.getString("News"));
                    fragment.setArguments(bundle);
                    FragmentTransaction transaction = getFragmentManager().beginTransaction();
                    transaction.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out, android.R.anim.fade_in, android.R.anim.fade_out);
                    transaction.replace(R.id.Container, fragment).addToBackStack(null).commit();
                }
            });
        }

        if(bundle.getString("Location") != null){
            mLocation.setVisibility(View.VISIBLE);
            mLocation.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(bundle.getString("Location"))));
                }
            });
        }

        if(bundle.getString("Twitter") != null){
            mCommunication.setVisibility(View.VISIBLE);
            Twitter.setVisibility(View.VISIBLE);
            Twitter.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    startActivity(new Intent(Intent.ACTION_VIEW , Uri.parse(bundle.getString("Twitter"))));
                }
            });
        }

        if(bundle.getString("Facebook") != null){
            mCommunication.setVisibility(View.VISIBLE);
            Facebook.setVisibility(View.VISIBLE);
            Facebook.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    startActivity(new Intent(Intent.ACTION_VIEW , Uri.parse(bundle.getString("Facebook"))));
                }
            });

        }
    }
}

File: app/src/main/java/com/ss/universitiesdirectory/ui/details/DetailsFragment.kt
package com.ss.universitiesdirectory.ui.details

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import androidx.navigation.fragment.navArgs
import coil.load
import com.ss.universitiesdirectory.R
import com.ss.universitiesdirectory.databinding.FragmentDetailsBinding
import com.ss.universitiesdirectory.utils.navigateTo
import com.ss.universitiesdirectory.utils.viewBinding

class DetailsFragment : Fragment(R.layout.fragment_details) {

    private val binding by viewBinding(FragmentDetailsBinding::bind)
    private val argument by navArgs<DetailsFragmentArgs>()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        showData()
    }

    private fun showData() {
        val university = argument.university
        val directions = DetailsFragmentDirections
        binding.detailsLogo.load(university.logo)
        binding.about.text = university.about

        if (university.colleges.isNotEmpty()) {
            binding.colleges.visibility = View.VISIBLE
            binding.colleges.setOnClickListener {
                val action = directions.actionDetailsFragmentToWebsiteFragment(university.colleges)
                findNavController().navigateTo(action, R.id.detailsFragment)
            }
        }
        if (university.news.isNotEmpty()) {
            binding.rss.visibility = View.VISIBLE
            binding.rss.setOnClickListener {
                val action = directions.actionDetailsFragmentToNewsFragment(university.news)
                findNavController().navigateTo(action, R.id.detailsFragment)
            }
        }
        if (university.website.isNotEmpty()) {
            binding.website.visibility = View.VISIBLE
            binding.website.setOnClickListener {
                val action = directions.actionDetailsFragmentToWebsiteFragment(university.website)
                findNavController().navigateTo(action, R.id.detailsFragment)
            }
        }
        if (university.application.isNotEmpty()) {
            binding.detailsApplication.visibility = View.VISIBLE
            binding.detailsApplication.setOnClickListener {
                startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(university.application)))
            }
        }
        if (university.location.isNotEmpty()) {
            binding.detailsLocation.visibility = View.VISIBLE
            binding.detailsLocation.setOnClickListener {
                startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(university.location)))
            }
        }
        if (university.twitter.isNotEmpty()) {
            binding.communicationHeader.visibility = View.VISIBLE
            binding.twitter.visibility = View.VISIBLE
            binding.twitter.setOnClickListener { openSocialMedia(university.twitter) }
        }
        if (university.facebook.isNotEmpty()) {
            binding.communicationHeader.visibility = View.VISIBLE
            binding.facebook.visibility = View.VISIBLE
            binding.facebook.setOnClickListener { openSocialMedia(university.facebook) }
        }
        if (university.youtube.isNotEmpty()) {
            binding.communicationHeader.visibility = View.VISIBLE
            binding.youtube.visibility = View.VISIBLE
            binding.youtube.setOnClickListener { openSocialMedia(university.youtube) }
        }
        if (university.instagram.isNotEmpty()) {
            binding.communicationHeader.visibility = View.VISIBLE
            binding.instagram.visibility = View.VISIBLE
            binding.instagram.setOnClickListener { openSocialMedia(university.instagram) }
        }
        if (university.snapchat.isNotEmpty()) {
            binding.snapchat.visibility = View.VISIBLE
            binding.communicationHeader.visibility = View.VISIBLE
            binding.snapchat.setOnClickListener { openSocialMedia(university.snapchat) }
        }
    }

    private fun openSocialMedia(stringUri: String) {
        val intent  = Intent(Intent.ACTION_VIEW, Uri.parse(stringUri))
        val chooser = Intent.createChooser(intent, "Open app")
        startActivity(chooser)
    }
}